### ISSUE TITLE
Schema: Do not allow an empty content in selected elements

### ISSUE THREAD

**Issue by matyaskopp** [clarin-eric/ParlaMint#354]
These elements should contain some content:
- `<desc/>`
- `<label/>`
- `<p/>`
- `<term/>`

I have seen it in PT corpus that passes the validation...

I guess this can be a common error in various corpora as everyone copies a header, remove old data and possibly doesn't add a new one, so we should handle this...

**Comment by TomazErjavec** [comment]
@matyaskopp, good point. However, there are some further questions here:

* What kind of content do we require? For example, having one space as the content obviously in not good enough, we actually want some non-whitespace content. But even so, shall we allow the string to start or end in whitespace? Shall we allow a series of whitespace characters in the string? Allow tabulators? New-lines? Ideally, no, to all these questions, but in this case existing corpora will most likely become invalid, and at this point it might be too late to impose new "ideal" restrictions, at least for V3.0. Btw. a discussion on this is on [Stack Overflow](https://stackoverflow.com/questions/6195885/define-an-element-as-non-empty-in-relaxng).
* The elements you list are just a few of the elements that are required to contain a (non-empty) string; in fact, there are probably only a few elements that have a choice of containing a string or being empty. So, we should probably extend this requirement to all the other "must contain text" elements. But, again, maybe too late for V3.

In short, I will try to implement "something" for the elements you mention, for the rest, I guess a new issue should be opened.

**Comment by TomazErjavec** [comment]
@matyaskopp, despite what I wrote, I did add the rather strict interpretation of what kind of string we accept, and not only for the elements you mention, but quite a few more in [be99bc2](https://github.com/clarin-eric/ParlaMint/commit/be99bc29e5c1d71c2f4f99accc887bcfed49cf52).

All these elements refer to https://github.com/clarin-eric/ParlaMint/blob/be99bc29e5c1d71c2f4f99accc887bcfed49cf52/Schema/ParlaMint-TEI.rng#L686-L692.

@starkadur, note that this new restriction breaks some of your already submitted files, but just a few, i.e. cases where your the text content of elements starts with a space, cf. [compile.log](https://github.com/clarin-eric/ParlaMint/blob/documentation/Schema/compile.log).

**Comment by TomazErjavec** [comment]
In [7e07dce](https://github.com/clarin-eric/ParlaMint/commit/7e07dcead80a6c42d5826ccf557b105b5a04781f) the "non-empty.string" has been renamed to have a more appropriate name (and documentation), it is now https://github.com/clarin-eric/ParlaMint/blob/7e07dcead80a6c42d5826ccf557b105b5a04781f/Schema/ParlaMint-TEI.rng#L684-L689.

@matyaskopp, note that I have also now applied this type of content to the name element, as it appears in the TEI header, so your data became invalid too, cf. the [complie.log](https://github.com/clarin-eric/ParlaMint/blob/documentation/Schema/compile.log). But I think it is easy to fix.

With this, I did all that I suggested in my [comment](https://github.com/clarin-eric/ParlaMint/issues/354#issuecomment-1273032463). Maybe too much, but the corpora will be better. I hope none of the partners freak out...

**Comment by TomazErjavec** [comment]
I  think we are done with this. Close?
